libbeat/publisher/queue/diskqueue/segments.go

```diff
@@ -424,25 +356,34 @@ func readSegmentHeader(in io.Reader) (*segmentHeader, error) {
 	if err != nil {
 		return nil, fmt.Errorf("could not read segment version: %w", err)
 	}
-
 	if header.version > currentSegmentVersion {
 		return nil, fmt.Errorf("unrecognized schema version %d", header.version)
 	}
-
 	if header.version >= 1 {
 		err = binary.Read(in, binary.LittleEndian, &header.frameCount)
 		if err != nil {
 			return nil, fmt.Errorf("could not read segment count: %w", err)
 		}
 	}
-	if header.version >= 2 {
-		err = binary.Read(in, binary.LittleEndian, &header.options)
-		if err != nil {
-			return nil, fmt.Errorf("could not read segment options: %w", err)
-		}
+	return header, nil
+}
+
+// writeSegmentHeader seeks to the beginning of the given file handle and
+// writes a segment header with the current schema version, containing the
+// given frameCount.
+func writeSegmentHeader(out *os.File, frameCount uint32) error {
+	_, err := out.Seek(0, io.SeekStart)
+	if err != nil {
+		return err
 	}
 
-	return header, nil
+	version := uint32(currentSegmentVersion)
+	err = binary.Write(out, binary.LittleEndian, version)
+	if err != nil {
+		return err
+	}
+	err = binary.Write(out, binary.LittleEndian, frameCount)
+	return err
```

Comment:
If we do end up sticking with schema 2 instead of incrementing to 3, then we should probably still write out null bytes for the options here.